privacyidea/lib/resolvers/LDAPIdResolver.py

```diff
@@ -188,28 +188,40 @@ def cache(func):
     """
     @functools.wraps(func)
     def cache_wrapper(self, *args, **kwds):
-        # If it does not exist, create the node for this instance
-        resolver_id = self.getResolverId()
-        if not resolver_id in CACHE:
-            CACHE[resolver_id] = {"getUserId": {},
-                                  "getUserInfo": {},
-                                  "_getDN": {}}
-
-        # get the portion of the cache for this very LDAP resolver
-        r_cache = CACHE.get(resolver_id).get(func.func_name)
-        if args[0] in r_cache and \
-                        datetime.datetime.now() < r_cache[args[0]][
-                    "timestamp"] + \
-                        datetime.timedelta(seconds=self.cache_timeout):
-            log.debug("Reading {0!r} from cache for {1!r}".format(args[0],
-                                                              func.func_name))
-            return r_cache[args[0]]["value"]
+        # Only run the code, in case we have a configured cache!
+        if self.cache_timeout > 0:
+            # If it does not exist, create the node for this instance
+            resolver_id = self.getResolverId()
+            now = datetime.datetime.now()
+            tdelta = datetime.timedelta(seconds=self.cache_timeout)
+            if not resolver_id in CACHE:
+                CACHE[resolver_id] = {"getUserId": {},
+                                      "getUserInfo": {},
+                                      "_getDN": {}}
+            else:
+                # Clean up the cache in the current resolver and the current function
+                _to_be_deleted = []
+                for user, cached_result in CACHE[resolver_id].get(func.func_name).iteritems():
```

Comment:
I'm not sure how this will perform in case we have a high-load server with a lot of cache entries, because now, we would iterate over all cached entries for every function invocation. But maybe that's not even noticeable, I don't know :-)

We could alternatively remove old entries only when we write new entries to the cache?

Reply:
We could create our own dataclass or use an external module. But for now we can not add an additional dependency in the patch release - imho.
@plettich Can you please add some ideas to an issue for milestone 3 / code cleanup?